tests/PhpSpreadsheetTests/Reader/Html/HtmlTest.php

```diff
@@ -363,4 +364,46 @@ public function testBorderWithColspan(): void
             self::assertEquals(Border::BORDER_THIN, $border->getBorderStyle());
         }
     }
+
+    public function testDataType(): void
+    {
+        $html = '<table>
+                    <tr>
+                        <td data-type="b">1</td>
+                        <td data-type="s">1234567</td>
+                        <td data-type="f">=CONCAT("TEXT A ","TEXT B")</td>
+                        <!-- in some cases, you may want to treat the string with beginning equal sign as a string rather than a formula -->
+                        <td data-type="s">=B1</td>
+                        <td data-type="d">2022-02-21 10:20:30</td>
+                        <td data-type="invalid-datatype">text</td>
+                    </tr>
+                </table>';
+
+        $reader = new Html();
+        $spreadsheet = $reader->loadFromString($html);
+        $firstSheet = $spreadsheet->getSheet(0);
+
+        // check boolean data type
+        self::assertEquals(DataType::TYPE_BOOL, $firstSheet->getCell('A1')->getDataType());
+        self::assertIsBool($firstSheet->getCell('A1')->getValue());
+
+        // check string data type
+        self::assertEquals(DataType::TYPE_STRING, $firstSheet->getCell('B1')->getDataType());
+        self::assertIsString($firstSheet->getCell('B1')->getValue());
+
+        // check formula data type
+        self::assertEquals(DataType::TYPE_FORMULA, $firstSheet->getCell('C1')->getDataType());
+
+        // check formula output
+        self::assertEquals('TEXT A TEXT B', $firstSheet->getCell('C1')->getFormattedValue());
+
+        // check string with beginning equal sign is string and not formula
+        self::assertEquals(DataType::TYPE_STRING, $firstSheet->getCell('D1')->getDataType());
+        self::assertEquals('=B1', $firstSheet->getCell('D1')->getValue());
+        self::assertIsString($firstSheet->getCell('D1')->getValue());
+        self::assertTrue($firstSheet->getCell('D1')->getStyle()->getQuotePrefix());
+
+        //check iso date
+        self::assertEquals($firstSheet->getCell('E1')->getValue(), 44613.43090277778);
```

Comment:
We sometimes see rounding problems. Please use assertEqualsWithDelta here; 1E-8 is probably a good enough delta.